pkg/util/hasher/hasher.go

```diff
@@ -65,13 +65,15 @@ func AddAnnotation(object metav1.Object) error {
 		annotations = map[string]string{}
 	}
 
-	objectHash, err := GenerateHash(object)
-	if err != nil {
-		return err
+	if _, ok := annotations[AnnotationHash]; !ok {
+		objectHash, err := GenerateHash(object)
+		if err != nil {
+			return err
+		}
+
+		annotations[AnnotationHash] = objectHash
```

Comment:
As far as I see in my cluster, this is setting the annotation on the STS metadata level, but it should be on the STS.ObjectMeta.Annotations level in order to work well. (work well = pods get restarted if the annotation changes)

Reply:
so you tested it and the pods that get created by the sts do not get annotated?